internal/driver/tempfile.go

```diff
@@ -21,6 +21,19 @@ import (
 	"sync"
 )
 
+// getNextTempFile returns the next available temp file in dir with the provided prefix and suffix.
+func getNextTempFile(dir, prefix, suffix string) (string, error) {
```

Comment:
There is quite a bit of duplicated code in here...  Can't you just use existing `newTempFile`, grab the name of the returned file using Name() and then Close() it before use?

Reply:
`newTempFile` creates the temp file. But this change is to skip it. I will modify `newTempFile` to use `getNextTempFile`.